src/core/thread/mesh_forwarder.cpp

```diff
@@ -1791,7 +1791,8 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame, ThreadError aError)
 void MeshForwarder::SetDiscoverParameters(uint32_t aScanChannels, uint16_t aScanDuration)
 {
     mScanChannels = (aScanChannels == 0) ? static_cast<uint32_t>(Mac::kScanChannelsAll) : aScanChannels;
-    mScanDuration = (aScanDuration == 0) ? static_cast<uint16_t>(Mac::kScanDurationDefault) : aScanDuration;
+    mScanDuration = (aScanDuration < static_cast<uint16_t>(Mac::kScanDurationDefault)) ?
+                    static_cast<uint16_t>(Mac::kScanDurationDefault) : aScanDuration;
```

Comment:
If `kScanDurationDefault` is the minimal allowed value, how about rename to `kMinimalScanDuration`.